### PR TITLE
Ports: Add Celeste Classic (ccleste) port

### PR DESCRIPTION
**Pull request by adamantike**
`ccleste` is a C source port of Celeste Classic. Game is currently playable.

serenity-screenshot

### CODE REVIEW

File: Ports/ccleste/patches/fix-data-paths.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/sdl12main.c b/sdl12main.c
+index db6d76d..3e2eb29 100644
+--- a/sdl12main.c
++++ b/sdl12main.c
+@@ -89,7 +89,7 @@ static char* GetDataPath(char* path, int n, const char* fname) {
+ #else
+ 	char pathsep = '/';
+ #endif //_WIN32
+-	snprintf(path, n, "data%c%s", pathsep, fname);
++	snprintf(path, n, "/opt/ccleste/data%c%s", pathsep, fname);
```
Comment (adamantike) [inline Ports/ccleste/patches/fix-data-paths.patch:10]: I'm open to suggestions here, if there are simple ways to make this more portable. For now, this gets the job done.

Comment (ADKaster) [reply Ports/ccleste/patches/fix-data-paths.patch:10]: you could set the launcher command to a generated (heredoc?) shell script that you also install, that does `cd /path/to/celeste`, similar to the half life 2 port

**Comment by EWouters** [comment]
Looks like a lot of fun!

**Comment by AtkinsSJ** [comment]
This has a conflict.

**Pull request closed without merging**